### PR TITLE
test: add compliance results baseline tests across four packages

### LINKED ISSUE #2084: Add comprehensive test baseline for compliance results before second rule engine lands
# Description

Several compliance-related packages in the results handling pipeline have little to no test coverage. This matters because the team is planning to add a second rule engine, and without a solid baseline it will be hard to verify that both engines produce consistent compliance results.

The gaps I found:

- `silentprinter.go` - 4 methods, 0 tests
- `summarytable.go` - 8 functions, 0 tests. This file directly formats compliance scores, control counters, and table headers shown to users
- `gotree/gotree.go` - several public methods like `AddTree`, `Text`, `Items` are completely untested
- `clusterscan.go` - constructor and no-op method untested

# Proposed fix

Add table-driven tests covering:

- All `SilentPrinter` methods verified as safe no-ops
- `ControlCountersForSummary` with 8 cases covering label correctness and counter values including edge cases
- `GetControlTableHeaders` in both short and full mode with exact column name assertions
- `GetColumnsAlignments` verifying all 5 column alignment configs
- `GenerateFooter` in both modes with exact compliance score format assertions
- `gotree` public methods covering empty trees, multiple children, subtree nesting, ordering, and nested Print output
- `NewClusterPrinter` and `PrintSummaryTable` no-op safety

### PR DESCRIPTION
**Pull request by Varadraj75**
## Overview

Several compliance-related packages had zero or minimal test coverage. With a second rule engine planned, this adds a baseline so regressions can be caught when both engines are compared.

test added across four packages:

- `silentprinter` - all four IPrinter methods verified as safe no-ops
- `summarytable` - ControlCountersForSummary, GetControlTableHeaders,
  GetColumnsAlignments, GenerateFooter with exact compliance score format assertions
- `gotree` - New, Add, AddTree, Text, Items, Print covering empty trees, nested subtrees, child ordering
- `clusterscan` - NewClusterPrinter and PrintSummaryTable no-op safety

## How to Test

```bash
go test -v -run "TestSilentPrinter|TestControlCounters|TestGetControlTable|TestGetColumns|TestGenerateFooter|TestNew|TestAdd|TestText|TestItems|TestPrint|TestNewCluster|TestClusterPrinter" \
  github.com/kubescape/kubescape/v3/core/pkg/resultshandling/printer/v2 \
  github.com/kubescape/kubescape/v3/core/pkg/resultshandling/printer/v2/prettyprinter/tableprinter/configurationprinter \
  github.com/kubescape/kubescape/v3/core/pkg/resultshandling/gotree
```

## Related issues/PRs

* Resolved #2084

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

* **Tests**
  * Expanded unit tests for tree construction/printing (creation, child/subtree insertion, accessors, print cases including empty root).
  * Added tests ensuring cluster printer and silent printer are safe no-ops (do not panic).
  * New comprehensive summary-table tests covering control counters, headers (short vs full), column alignments, and footers with exact formatting and counts.

[Review Change Stack](https://app.coderabbit.ai/change-stack/kubescape/kubescape/pull/2085)

### CODE REVIEW

File: core/pkg/resultshandling/gotree/gotree_test.go [hunk 1]
```diff
@@ -169,3 +170,140 @@ func Test_printer_printItems(t *testing.T) {
...
+}
+
+func TestAddTree_MultipleSubtrees(t *testing.T) {
+	root := New("root")
+	s1 := New("s1")
+	s2 := New("s2")
+	root.AddTree(s1)
+	root.AddTree(s2)
+	assert.Len(t, root.Items(), 2)
+}
```
[the remark this answers is not in the document]
Comment (Varadraj75) [reply core/pkg/resultshandling/gotree/gotree_test.go:242]: Fixed, TestAddTree_MultipleSubtrees now asserts both subtree order  and count. TestAddTree_PreservesSubtreeItems now validates parent  slice length before indexing.

File: core/pkg/resultshandling/gotree/gotree_test.go [hunk 2]
```diff
@@ -169,3 +170,140 @@ func Test_printer_printItems(t *testing.T) {
...
+
+func TestAddTree_PreservesSubtreeItems(t *testing.T) {
+	root := New("root")
+	subtree := New("sub")
+	subtree.Add("a")
+	subtree.Add("b")
+	root.AddTree(subtree)
+	require.Len(t, root.Items()[0].Items(), 2)
+	assert.Equal(t, "a", root.Items()[0].Items()[0].Text())
+	assert.Equal(t, "b", root.Items()[0].Items()[1].Text())
```
[the remark this answers is not in the document]
Comment (Varadraj75) [reply core/pkg/resultshandling/gotree/gotree_test.go:312]: Fixed, TestAddTree_MultipleSubtrees now asserts both subtree order  and count. TestAddTree_PreservesSubtreeItems now validates parent  slice length before indexing.

File: core/pkg/resultshandling/printer/v2/prettyprinter/tableprinter/configurationprinter/summarytable_test.go [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+type mockCounters struct {
+	failed, skipped, passed, excluded int
+}
+
+func (m mockCounters) Failed() int   { return m.failed }
+func (m mockCounters) Skipped() int  { return m.skipped }
+func (m mockCounters) Passed() int   { return m.passed }
+func (m mockCounters) Excluded() int { return m.excluded }
+func (m mockCounters) All() int      { return m.failed + m.skipped + m.passed + m.excluded }
```
Comment (matthyx) [inline core/pkg/resultshandling/printer/v2/prettyprinter/tableprinter/configurationprinter/summarytable_test.go]: Blocking: this mock does not match the real `reportsummary.ICounters` implementations. `StatusCounters.All()` and `PostureCounters.All()` both sum only failed + passed + skipped; `Excluded()` is deprecated and is not included. With the current mock, the `with excluded` case is asserting behavior production counters can never produce. Please change `mockCounters.All()` to exclude `excluded`, and update or remove the `with excluded` test case (`wantAll` should be `3`, not `6`).

Comment (Varadraj75) [reply core/pkg/resultshandling/printer/v2/prettyprinter/tableprinter/configurationprinter/summarytable_test.go]: Fixed, mockCounters.All() now sums only failed + skipped + passed, matching StatusCounters and PostureCounters. The test case name and wantAll updated accordingly (3, not 6).

**Review by matthyx — APPROVED** [review]
thanks @Varadraj75
